share/functions/__fish_preview_current_file.fish

```diff
@@ -0,0 +1,16 @@
+function __fish_preview_current_file --description "Open the current file in a pager"
+    # set pager to user-preferred pager or less
+    set -l pager "less --"
+    set -q PAGER && echo $PAGER | read -at pager
+
+    # get current or last token, use eval to resolve $HOME and ~
+    set -l file (eval echo (commandline -t))
+    test -z "$file" && set -l file (eval echo (commandline -oc)[-1])
+
+    # strip -option= from token if necessary
+    not test -e "$file" && set -l file (eval echo (string replace -r -- '^-[^\s=]*=(.*)' '$1' "$file"))
```

Comment:
I think this `eval` can be dropped: `$file` is already expanded, and you are only removing a prefix.

Reply:
Unfortunately not. The `~` only gets expanded when at the beginning of the token. That's not the case when eval runs first. Test for example `eval echo "--long-opt=~/Desktop/foo"`. Hence the second run after removing the options.
I will incorporate your other suggestions tomorrow.

Reply:
Oh yeah you're right. Maybe we should do the replacement before eval.